src/sidebar/services/groups.js

```diff
@@ -137,9 +137,40 @@ function groups($rootScope, store, api, isSidebar, localStorage, serviceUrl, ses
         params.document_uri = uri;
       }
       documentUri = uri;
-
-      // Fetch groups from the API.
-      return api.groups.list(params, null, { includeMetadata: true });
+      
+      if (features.flagEnabled('community_groups')) {
+        return $q.all([api.profile.groups.read({expand: "organization"}),
+               api.groups.list(params, null, { includeMetadata: true })]
+        ).then(data => {
+          const myGroups = data[0];
+          const listGroups = data[1].data;
+          let groups = [];
+          var i;
+  
+          const myGroupIds = [];
+          for(var i=0; i<myGroups.length; ++i) {
+            myGroupIds.push(myGroups[i].id);
+            myGroups[i].groupList = "myGroups";
+          }
+  
+          const featuredGroups = [];
+          for(var i=0; i<listGroups.length; ++i) {
+            if (listGroups[i].id === '__world__') {
+              listGroups[i].groupList = "myGroups";
+              groups.push(listGroups[i]);
+            }
+            else if (myGroups.includes(listGroups[i].id)) {
+              featuredGroups.push(listGroups[i]);
+            }
+          }
+  
+          groups = groups.concat(myGroups).concat(featuredGroups)
+          return {data: groups, token: featuredGroups.token};
```

Comment:
Might want to pause and think about the shape of the returned value here. Returning a single Array will certainly match what's returned by the `load` method currently, but leaves you with the need to later iterate over your single Array and look at the `groupList` property.

Alternately you could return an object, like so:

```
{
  featuredGroups: [ ... ],
  myGroups: [ ... ],
}
```

though that would require re-jiggering a few other spots. I respect why you'd do this in a flat list...I just wanna be sure it's the right choice now!

Reply:
Yeah, I did that specifically so I didn't have to re-jigger anything. It's a simple filter in the template when presenting to the user so figured it wouldn't be such a big deal but certainly it could go the other route too. It does take up more memory this way, particularly as the group list grows, on the other hand it doesn't require any changes to get or leave group methods and keeps the presentation/ux for the most part out of the way of the client logic which I like.